lib/dachsfisch/json2_xml_converter.rb

```diff
@@ -5,17 +5,17 @@ class JSON2XMLConverter < ConverterBase
     def initialize(json:)
       super()
       @json_hash = JSON.parse json
-      if @json_hash.length > 1
-        raise InvalidJSONInputError.new('multiple root nodes are not supported')
-      end
     rescue TypeError, JSON::ParserError => e
       raise InvalidJSONInputError.new(e.message)
     end
 
     def execute
-      Nokogiri::XML::Builder.new do |xml|
+      fragment = Nokogiri::XML::DocumentFragment.parse ''
+
+      Nokogiri::XML::Builder.with fragment do |xml|
         add_element xml, @json_hash
-      end.to_xml
+      end
+      fragment.elements.deconstruct.map(&:to_xml).join("\n")
```

Comment:
Why aren't you just calling `fragment.to_xml` here? When reading the docs, I mainly see two differences (between your code and my question):

1. Different root nodes are separated by `'\n'` rather than `''` (a minor change, I'd say).
2. Text elements in the root level are stripped (this might be relevant). Check with the following example and check for `evil`: 

    ```rb
    def self.json
      <<~JSON
        {
          "alice": {
            "$1": "bob"
          },
          "$1": "evil",
          "charlie": {
            "$1": "david"
          }
        }
      JSON
    end
    ```

Your choice might be fine, I am just trying to understand it.

Reply:
I did indeed change it add a `\n` between nodes. Did you see another way to change the separators?

Reply:
Regarding point 2:
I don't think we need to support a text-node without parent

```XML
<alice>bob</alice>
evil
<charlie>david</charlie>
```
doesn't feel right imo

Reply:
> Did you see another way to change the separators?

No, not really.

> doesn't feel right imo

Fine for me.